### PR TITLE
Revise document for handling exceptions

### PR DESCRIPTION
**Pull request by brfrn169**
This PR revises the document for handling exceptions. Please take a look!

### CODE REVIEW

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: docs/api-guide.md [hunk 1]
```diff
@@ -786,107 +786,115 @@ Let's look at the following example code to see how to handle exceptions in Scal
...
+If you can this exception, it indicates that the condition for the mutation operation is not met.
+You can handle this exception according to your application requirements.
 
 Also, the `commit()` API could throw `CommitException`, `CommitConflictException`, and `UnknownTransactionStatusException`.
-If you catch `CommitException`, like the `CrudException` case, you should cancel the transaction or retry the transaction after the failure or error is fixed.
-If you catch `CommitConflictException`, like the `CrudConflictException` case, you can retry the transaction from the beginning.
-If you catch `UnknownTransactionStatusException`, you are not sure if the transaction succeeds or not.
+If you catch `CommitException`, it indicates that committing the transaction fails due to transient or nontransient faults. You can try retrying the transaction from the beginning, but the transaction may still fail if the cause is nontranient.
+If you catch `CommitConflictException`, it indicates that committing the transaction fails due to transient faults (e.g., a conflict error). You can retry the transaction from the beginning.
```
Comment (josh-wong) [inline docs/api-guide.md:878]:
```suggestion
If you catch `CommitConflictException`, this indicates that committing the transaction has failed due to transient faults (e.g., a conflict error). You can retry the transaction from the beginning.
```

File: docs/api-guide.md [hunk 2]
```diff
@@ -786,107 +786,115 @@ Let's look at the following example code to see how to handle exceptions in Scal
...
+If you catch `UnknownTransactionStatusException`, it indicates that you are not sure if the transaction succeeds or not.
```
Comment (josh-wong) [inline docs/api-guide.md:879]:
```suggestion
If you catch `UnknownTransactionStatusException`, this indicates that you are not sure if the transaction succeeds or not.
```

Although I edited the above sentence a bit, it is a bit unclear to me. Should `this indicates that you are not sure if the transaction succeeds or not` be something like `this indicates that the status of the transaction, whether it has succeeded or not, is unknown` instead?

File: docs/api-guide.md [hunk 3]
```diff
@@ -786,107 +786,115 @@ Let's look at the following example code to see how to handle exceptions in Scal
...
+
+In the sample code, for `UnknownTransactionStatusException`, the transaction doesn't retry because the cause of the exception is nontransient.
+Also, for `UnsatisfiedConditionException`, the transaction doesn't retry because how to handle this exception depends on your application requirements.
+For other exceptions, the transaction tries retrying because the cause of the exception is transient or nontransient.
+If the cause of the exception is transient, the transaction may succeed if you retry it.
+However, if the cause of the exception is nontransient, the transaction may still fail even if you retry it.
+In such a case, you will exhaust the number of retries.
+
+Please note that if you begin a transaction by specifying a transaction ID, you must use a different ID when you retry the transaction.
+And, in the sample code, the transaction retries three times maximum and sleeps fixed 100 milliseconds before it retries.
```
Comment (josh-wong) [inline docs/api-guide.md:896]:
```suggestion
And, in the sample code, the transaction retries three times maximum and sleeps for 100 milliseconds before it retries.
```

File: docs/api-guide.md [hunk 4]
```diff
@@ -786,107 +786,115 @@ Let's look at the following example code to see how to handle exceptions in Scal
...
+But you can choose a retry policy according to your application requirements, such as exponential backoff.
```
Comment (josh-wong) [inline docs/api-guide.md:897]:
```suggestion
But you can choose a retry policy, such as exponential backoff, according to your application requirements.
```

**Review by josh-wong — CHANGES_REQUESTED** [review]
I've added a few comments and suggestions. PTAL!

File: docs/api-guide.md (hunk 2 above)
Comment (brfrn169) [reply docs/api-guide.md]: Sounds good. Thank you! Fixed in 0b33081cfbc4f37516948c8d6c4b84821c2f0a29.

**Review by josh-wong — APPROVED** [review]
LGTM! Thank you!🙇‍♂️

File: docs/api-guide.md [hunk 5]
```diff
@@ -786,107 +786,115 @@ Let's look at the following example code to see how to handle exceptions in Scal
...
-      } catch (CrudException | CommitException e) {
-        // If you catch CrudException or CommitException, it indicates some failure happens, so you
-        // should cancel the transaction or retry the transaction after the failure or error is 
-        // fixed
+        // You need to handle `UnsatisfiedConditionException` only if a mutation operation specifies
+        // a condition. This exception indicates the condition for the mutation operation is not met
+
         try {
-          tx.rollback();
+          transaction.rollback();
```
Comment (komamitsu) [inline docs/api-guide.md:826]: [trivial] I guess those who read the error handling for `rollback` might wonder why just logging is okay. Maybe describing the reason that the failures of `rollback` can be ignored would be helpful for users.

Comment (brfrn169) [reply docs/api-guide.md:826]: Thank you. I added some more explanation in ad4828e21cf86f30b1eae84b30aed1c8eb0653d8. Please take a look when you have time!

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by komamitsu — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**